PhysicsTools/NanoAOD/plugins/GenPartIsoProducer.cc

```diff
@@ -0,0 +1,163 @@
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/JetReco/interface/GenJet.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+
+#include "DataFormats/HepMCCandidate/interface/GenParticleFwd.h"
+#include "SimDataFormats/GeneratorProducts/interface/GenEventInfoProduct.h"
+#include "DataFormats/PatCandidates/interface/PackedGenParticle.h"
+#include "DataFormats/PatCandidates/interface/CompositeCandidate.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+
+#include <tuple>
+#include <string>
+#include <vector>
+#include <TLorentzVector.h>
+
+using namespace std;
+
+class GenPartIsoProducer : public edm::stream::EDProducer<> {
+public:
+  explicit GenPartIsoProducer(const edm::ParameterSet& iConfig)
+      : finalGenParticleToken(consumes<reco::GenParticleCollection>(iConfig.getParameter<edm::InputTag>("src"))) {
```

Comment:
please replace `src` with `genPart` and make the second input tag configurable, like 'packedGenPart` instead of hard coding it.

Reply:
Done, addressed in my last commit.